### PR TITLE
added initial code for integrity sum project

### PR DESCRIPTION
**Pull request by Kaborda-Irina**

### CODE REVIEW

File: internal/core/services/hash_service.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+package services
```
Comment (vangithub1992) [inline internal/core/services/hash_service.go:1]: If we already named package `services`, we shouldn't use `_service` postfix in file names, so I'm advising to remove from file names postfixes `_service` and `_ports`

File: internal/core/services/service.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
+package services
```
Comment (vangithub1992) [inline internal/core/services/service.go:1]: And this may be renamed to `app`, not a basis service

File: internal/repositories/connection.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	connectionDB := models.ConnectionDB{
+		Dbdriver:   os.Getenv("DB_DRIVER"),
+		DbUser:     os.Getenv("DB_USER"),
+		DbPassword: os.Getenv("DB_PASSWORD"),
+		DbPort:     os.Getenv("DB_PORT"),
+		DbHost:     os.Getenv("DB_HOST"),
+		DbName:     os.Getenv("DB_NAME"),
+	}
+	if connectionDB.Dbdriver == "" {
+		logger.Error("DB_DRIVER was not set")
```
Comment (vangithub1992) [inline internal/repositories/connection.go:23]: we can `return errors.New("DB_DRIVER was not set")` here. 
or we can log a little more data and do some actions. like 
`
logger.Errorf("DB_DRIVER was not set. setting by default: %s",  defaultValue)
`

Comment (vangithub1992) [reply internal/repositories/connection.go:23]: or we can validate needed for us env values before creating a connection to DB. On the stage of setting env values to environment

File: manifests/database/postgres-db-pv.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    app: kuber-hasher
+spec:
+  storageClassName: manual                    # Storage class. A PV Claim requesting the same storageClass can be bound to this volume.
+  capacity:
+    storage: 5Gi
+  accessModes:
+    - ReadWriteMany
+  hostPath:                                   # hostPath PersistentVolume is used for development and testing. It uses a file/directory on the Node to emulate network-attached storage
+    path: "/mnt/data"
+  persistentVolumeReclaimPolicy: Retain       # Retain the PV even after PVC is deleted.
```
Comment (vangithub1992) [inline manifests/database/postgres-db-pv.yaml:16]: No new line before EOF )

You can check it by yourself on the github )
Screenshot 2022-07-21 at 12 19 43

File: pkg/logger/logger.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+// Formatter implements logrus.Formatter interface.
+type formatter struct {
+	prefix string
+}
+
+// Format building log message.
+func (f *formatter) Format(entry *logrus.Entry) ([]byte, error) {
+	var sb bytes.Buffer
+
+	var newLine = "\n"
```
Comment (vangithub1992) [inline pkg/logger/logger.go:43]: I don't think we need to check which OS is used by the runtime. We already specified our OS in the docker file. I think that if a user wants to change the OS preferences in the project by changing OS in the docker file, he'll be in need of changing some additional features, and this line also, cause we skip a lot of GOOS types, also macOS is supporting Linux `\r\n`.

File: internal/core/services/service.go (hunk 2 above)
Comment (Kaborda-Irina) [reply internal/core/services/service.go:1]: Fixed, renamed.

File: internal/core/services/hash_service.go (hunk 1 above)
Comment (Kaborda-Irina) [reply internal/core/services/hash_service.go:1]: Fixed, renamed.

File: manifests/database/postgres-db-pv.yaml (hunk 4 above)
Comment (Kaborda-Irina) [reply manifests/database/postgres-db-pv.yaml:16]: Fixed, added new line before EOF

File: pkg/logger/logger.go (hunk 5 above)
Comment (Kaborda-Irina) [reply pkg/logger/logger.go:43]: OK. I understand

Comment (Kaborda-Irina) [reply pkg/logger/logger.go:43]: had to leave. 
`var newLine = "\n"
	if runtime.GOOS == "linux" {
		newLine = "\r\n"
	}
`
because it does not display logs

Comment (vangithub1992) [reply pkg/logger/logger.go:43]: If we set `newLine = "\n"` or `newLine = "\r\n"` ?

File: internal/repositories/connection.go (hunk 3 above)
Comment (Kaborda-Irina) [reply internal/repositories/connection.go]: https://github.com/ScienceSoft-Inc/integrity-sum/pull/1/commits/ac67a41c7f995d340d28d679575722c33bdf935a#diff-d8b26157f5d098431f11d44d50e4432702510f6044b42dca6390530b90e66377
created new function. It's a normal or not?

File: pkg/logger/logger.go (hunk 5 above)
Comment (Kaborda-Irina) [reply pkg/logger/logger.go:43]: I understood, I just deleted ` newlines`

File: .env [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+#number of running workers in the workerpool
+COUNT_WORKERS=4
+
+PROC_DIR="/proc"
+
+#hashing algorithm for hashing data
+ALGORITHM="SHA256"
+
+#configMap name for hasher
+CONFIG_MAP_NAME_FOR_HASHER=integrity-sum-config
```
Comment (vangithub1992) [inline .env:25]: Did not looks informative to me. Maybe we can add additional comments(like where it is used) to this parameter or change its name?

File: README.md [hunk 7]
```diff
@@ -1 +1,122 @@
...
+* SHA512
+
+## Architecture
+
+## Getting Started
+
+### Clone repository and install dependencies
+```
+$ git clone https://github.com/ScienceSoft-Inc/integrity-sum.git
+$ cd path/to/install
```
Comment (vangithub1992) [inline README.md:34]: Should we add here `go mod download` to ensure that the user has needed libraries from `go.mod`?

File: internal/core/services/k8s.go [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+func (ks *KuberService) GetDataFromDeployment(kuberData *models.KuberData) (*models.DeploymentData, error) {
+	allDeploymentData, err := kuberData.Clientset.AppsV1().Deployments(kuberData.Namespace).Get(context.Background(), kuberData.TargetName, metav1.GetOptions{})
+	if err != nil {
+		ks.logger.Error("err while getting data from kuberAPI ", err)
+		return nil, err
+	}
+
+	deploymentData := models.DeploymentData{}
```
Comment (vangithub1992) [inline internal/core/services/k8s.go:134]: Could we change it to an "inline" declaration? )
like `
deploymentData := models.DeploymentData{
NameDeployment = kuberData.TargetName,
Timestamp = fmt.Sprintf("%v", allDeploymentData.CreationTimestamp),
NamePod = os.Getenv("POD_NAME")
}
`

File: pkg/api/hasher.go [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+		select {
+		case hashData, ok := <-results:
+			if !ok {
+				return allHashData
+			}
+			allHashData = append(allHashData, hashData)
+		case <-c:
+			fmt.Println("exit program")
+			return []*HashData{}
+		case <-ctx.Done():
```
Comment (vangithub1992) [inline pkg/api/hasher.go:45]: Should we return something if we catch `ctx.Done()`?

**Review by vangithub1992 — CHANGES_REQUESTED** [review]
What about `.editorconfig` file?

File: .env (hunk 6 above)
Comment (Kaborda-Irina) [reply .env:25]: fixed it

File: README.md (hunk 7 above)
Comment (Kaborda-Irina) [reply README.md:34]: Fixed it.

File: internal/core/services/k8s.go (hunk 8 above)
Comment (Kaborda-Irina) [reply internal/core/services/k8s.go:134]: Fixed it

File: pkg/api/hasher.go (hunk 9 above)
Comment (Kaborda-Irina) [reply pkg/api/hasher.go:45]: fixed it

File: .editorconfig [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+indent_size = 4
+
+[*.md]
+indent_size = 4
+trim_trailing_whitespace = false
+
+eclint_indent_style = unset
+
+[Dockerfile]
+indent_size = 4
```
Comment (vangithub1992) [inline .editorconfig:21]: New line at EOF, please 🌚

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+*.so
+*.dylib
+
+# Test binary, built with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
```
Comment (vangithub1992) [inline .gitignore:14]: also go developers often use vscode.

File: cmd/main.go [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	"os/signal"
+
+	"github.com/integrity-sum/internal/initialize"
+	"github.com/joho/godotenv"
+)
+
+func main() {
+	// Load values from .env into the system
+	if err := godotenv.Load(); err != nil {
+		log.Print("No .env file found")
```
Comment (vangithub1992) [inline cmd/main.go:18]: should we stop executing if the .env file was not found?

**Review by vangithub1992 — APPROVED** [review]
Approve but still need some code style changes

File: .editorconfig (hunk 10 above)
Comment (Kaborda-Irina) [reply .editorconfig:21]: fixed it

File: cmd/main.go (hunk 12 above)
Comment (Kaborda-Irina) [reply cmd/main.go:18]: Ok. Understood. Fixed it

File: .gitignore (hunk 11 above)
Comment (Kaborda-Irina) [reply .gitignore:14]: added .vscode in .gitignore

**Review by Denislite — APPROVED** [review]
LGTM

**Review by TimofeiShorokhov — APPROVED** [review]

**Pull request merged**